components/script/dom/htmlanchorelement.rs

```diff
@@ -667,7 +682,23 @@ pub fn follow_hyperlink(subject: &Element, hyperlink_suffix: Option<String>) {
         };
 
         // Step 14
-        debug!("following hyperlink to {}", url);
-        target_window.load_url(url, replace, false, referrer, referrer_policy);
+        let pipeline_id = target_window.upcast::<GlobalScope>().pipeline_id();
+        let load_data = LoadData::new(
+            LoadOrigin::Script(document.url().origin()),
```

Comment:
Should this be the origin associated with the document's URL, or the document's domain?

Reply:
I think the document's domain is better, I switched to using that, thanks!